tests/main/views/test_brief_response.py

```diff
@@ -605,19 +605,15 @@ def test_can_not_submit_a_brief_response_for_a_framework_that_is_not_live_or_exp
                 db.session.delete(existing_brief_response[-1])
                 db.session.commit()
 
-            # Get dos framework.
-            dos_framework = db.session.query(Framework).filter_by(slug='digital-outcomes-and-specialists').first()
-
-            # Make framework live so a brief response can be created.
-            dos_framework.status = 'live'
-            db.session.commit()
-
+            # Ensure dos framework is live.
+            db.session.query(Framework).filter_by(slug='digital-outcomes-and-specialists').update({'status': 'live'})
```

Comment:
I don't understand why an 'update' is possible but a fetch-change-commit is not possible?

Reply:
So this line here:
https://github.com/alphagov/digitalmarketplace-api/pull/719/files#diff-62b9141c1c9c33809d1460a992905216L619
`dos_framework.status = framework_status`
Was the line operating on a stale object.

To successfully update that object I would have had to repeat the lines above, first getting, then setting the attr, then committing the action. 

Then I remembered 'there's a shortcut for that'.

So I used update. But then I thought I might as well standardise both operations.

Reply:
Ah hmmm but you're kinda doing the wrong thing then, aren't you: the intention of the code is presumably to _reset_ the entire object back how it _was_ - including the status. So using a 'stale' object was _correct_, no? 

Just updating the one field might not put things back exactly how they used to be?

Reply:
That _may_ have been the intention :man_shrugging: 
I don;t think we can assume so because:
1. The `_setup_existing_brief_response` shouldn't _change_ the framework object
2. We're only updating the framework so we _can_ create the objects for this test therefore we only care about slug, status and maybe family.

Regardless of the intention:
1. SQLAlchemy doesn't commit the whole object on a single attribute change just the changed attributes with an `UPDATE  SET WHERE id =` command
2. That line isn't actually doing anything at all. The live_dos_framework fixture is being used here too. And SQLAlchemy won't perform a useless operation (ie issuing an update that doesn't change anything, updating status to 'live' when it's already 'live').
3. Also rather than updating the whole object back to how it was issuing an update on a stale object doesn't do _anything_

Reply:
Removed the first one 👍